### ISSUE TITLE
Flashing a known app crashes the CLI

### ISSUE THREAD

**Issue by suda** [spark/particle-cli#331]
Running:

`$ particle flash MY_DEVICE tinker`

crashes with:

```
attempting to flash firmware to your device MY_DEVICE
Flash device failed
TypeError: Cannot convert undefined or null to object
    at Object._addFilesToCompile (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/oldlib/ApiClient.js:537:21)
    at Object.flashDevice (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/oldlib/ApiClient.js:474:8)
    at flashyFlash (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/commands/CloudCommands.js:382:16)
    at runTask (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/pipeline.js:34:12)
    at /Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/pipeline.js:42:12
    at tryCatchResolve (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/lib/apply.js:46:23)
    at callAndResolve (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/lib/apply.js:30:12)
    at callAndResolveNext (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/lib/apply.js:40:4)
    at Fold.fulfilled (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/lib/makePromise.js:767:11)
    at tryCatchReject (/Users/suda/.nvm/versions/node/v6.9.4/lib/node_modules/particle-cli/node_modules/when/lib/makePromise.js:845:30)
```

**Comment by hwestbrook** [comment]
I have this same issue. If someone else comes here looking for a fix, the fix is to downgrade the cli to v1.18.0 via `npm install -g particle-cli@1.18.0`

**Comment by m-mcgowan** [comment]
The problem is caused by changes to the api.flashDevice function, which can take an optional map of logical filenames to actual filenames. Not all callers provide the map (in particular, flashing a binary), so I had added code to generate a identity map from the list of files. However, this code was removed...
